### PR TITLE
bootloader/assets, tests: add factory-reset mode, test non-encrypted factory-reset

### PR DESCRIPTION
**Pull request by bboozzoo**
Based on top of #11571, adds the factory-reset mode to recovery grub boot script and a spread test to execute factory-reset on a non-encrypted system

### CODE REVIEW

**Comment by bboozzoo** [comment]
Yay, the spread test did not fail

**Review by pedronis — APPROVED** [review]
thank you

File: tests/nested/core/core20-factory-reset/task.yaml [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+    retry -n 60 --wait 2 tests.nested exec snap model --serial
+
+    # post factory reset world
+
+    tests.nested exec snap model > current-model
+    tests.nested exec snap model --serial > current-serial
+    tests.nested exec sudo blkid |grep -v /dev/loop > current-disk
+    # serials should be identical
+    diff -up initial-model current-model
+    diff -up initial-serial current-serial
```
Comment (mardy) [inline tests/nested/core/core20-factory-reset/task.yaml:52]: The `-p` option seems out of place, unless I miss something.

**Review by mardy — APPROVED** [review]
Excellent, only nitpicks in case you need to touch the code again.

File: tests/nested/core/core20-factory-reset/task.yaml (hunk 1 above)
Comment (mvo5) [reply tests/nested/core/core20-factory-reset/task.yaml]: I removed the -p now

**Comment by bboozzoo** [comment]
This is currently blocked by  https://github.com/canonical/ubuntu-image/pull/47

**Comment by jawn-smith** [comment]
The ubuntu-image PR has been merged.

**Pull request merged**